feat(blockdata): Add LadderDataMock with unit tests

thelooter opened the pull request:
- Implemented `LadderDataMock` for `Ladder` block data.
- Created corresponding unit tests in `LadderDataMockTest`.
- Added support for `facing`, `waterlogged` properties and cloning functionality.

# Description

<!-- State the changes of the pull request below. -->

# Checklist

The following items should be checked before the pull request can be merged.

- [ ] Code follows existing style.
- [ ] Unit tests added (if applicable).

thelooter committed:
feat(blockdata): Add LadderDataMock with unit tests

- Implemented `LadderDataMock` for `Ladder` block data.
- Created corresponding unit tests in `LadderDataMockTest`.
- Added support for `facing`, `waterlogged` properties and cloning functionality.

Signed-off-by: thelooter <evekolb2204@gmail.com>

4everTheOne commented on src/main/java/org/mockbukkit/mockbukkit/block/data/LadderDataMock.java:
```diff
@@ -0,0 +1,74 @@
+	 *
+	 * @param other the other block data.
+	 */
+	public LadderDataMock(@NotNull BlockDataMock other)
```
The constructor should only accept the LadderDataMock

4everTheOne reviewed (commented):
Looks good to me, just have a small suggestion

thelooter committed:
fix(blockdata): Correct LadderDataMock constructor signature

- Updated `LadderDataMock` constructor to accept the narrower `LadderDataMock` parameter type instead of the generic `BlockDataMock`.
- Removed redundant newlines in `LadderDataMockTest`.

Signed-off-by: thelooter <evekolb2204@gmail.com>

thelooter committed:
test(blockdata): Change LadderDataMockTest to use package-private class modifier

Signed-off-by: thelooter <evekolb2204@gmail.com>

4everTheOne committed:
Merge branch 'v1.21' into feat/ladder
